tests/models/test_token_classification_with_seq2seq_encoder_and_crf.py

```diff
@@ -0,0 +1,258 @@
+                torch.rand(self.batch_size, self.seq_len, self.hidden_size)
+            )
+            return BaseModelOutputWithPoolingAndCrossAttentions(
+                last_hidden_state=last_hidden_state,
+            )
+
+    hidden_size = 10
+
+    monkeypatch.setattr(
+        transformers.AutoConfig,
+        "from_pretrained",
+        lambda model_name_or_path: MockConfig(
+            hidden_size=hidden_size, classifier_dropout=0.1, model_type=model_type
+        ),
+    )
+    monkeypatch.setattr(
+        transformers.AutoModel,
+        "from_pretrained",
+        lambda model_name_or_path, config: MockModel(
+            batch_size=batch_size,
+            seq_len=seq_len,
+            hidden_size=hidden_size,
+            add_dummy_linear=add_dummy_linear,
+        ),
+    )
+
+    # set seed to make the classifier deterministic
+    torch.manual_seed(42)
+    result = TokenClassificationModelWithSeq2SeqEncoderAndCrf(
+        model_name_or_path="bert",
+        num_classes=num_classes,
+        warmup_proportion=0.0,
+        **model_kwargs,
+    )
+    assert not result.is_from_pretrained
+
+    return result
+
+
+@pytest.fixture
+def model(monkeypatch, batch):
+    inputs, targets = batch
+    model = get_model(
+        monkeypatch=monkeypatch,
+        model_type="bert",
+        batch_size=inputs["input_ids"].shape[0],
+        seq_len=inputs["input_ids"].shape[1],
+        num_classes=int(torch.max(targets) + 1),
+    )
+    # model = TokenClassificationModelWithSeq2SeqEncoderAndCrf(
+    #     model_name_or_path="bert-base-uncased",
+    #     num_classes=5,
+    # )
+    return model
+
+
+@pytest.mark.parametrize("model_type", ["bert", "albert", "distilbert", "longformer"])
+def test_config_model_classifier_dropout(monkeypatch, model_type):
+    model = get_model(
+        monkeypatch=monkeypatch,
+        model_type=model_type,
+        batch_size=4,
+        seq_len=10,
+        num_classes=5,
+    )
+    assert model is not None
+
+
+def test_freeze_base_model(monkeypatch, batch):
+    inputs, target = batch
+    # set seed to make the classifier deterministic
+    model = get_model(
+        monkeypatch,
+        model_type="bert",
+        batch_size=4,
+        seq_len=10,
+        num_classes=5,
+        add_dummy_linear=True,
+        freeze_base_model=True,
+    )
+    base_model_params = list(model.model.parameters())
+    # the dummy linear from the mock base model has 2 parameters
+    assert len(base_model_params) == 2
+    for param in base_model_params:
+        assert not param.requires_grad
+
+
+def test_forward(batch, model):
+    inputs, targets = batch
+    batch_size, seq_len = inputs["input_ids"].shape
+    num_classes = int(torch.max(targets) + 1)
+
+    # set seed to make sure the output is deterministic
+    torch.manual_seed(42)
+    output = model.forward(inputs)
+    assert set(output) == {"logits"}
+    logits = output["logits"]
+    assert logits.shape == (batch_size, seq_len, num_classes)
```

Comment:
test the actual values in logits (with `torch.testing.assert_close`, see the tests in `test_pooler.py`)